gobblin-compaction/src/main/java/org/apache/gobblin/compaction/action/CompactionHiveRegistrationAction.java

```diff
@@ -57,12 +58,14 @@
 
   private final State state;
   private EventSubmitter eventSubmitter;
+  private InputRecordCountHelper helper;
 
   public CompactionHiveRegistrationAction (State state) {
     if (!(state instanceof WorkUnitState)) {
       throw new UnsupportedOperationException(this.getClass().getName() + " only supports workunit state");
     }
     this.state = state;
+    this.helper = new InputRecordCountHelper(state);
```

Comment:
Do you need to instantiate this object here and call readExecutionCount or this can be passed down through State object ?  If you prefer doing this, we could lazily init this object whenever required to be used.

Reply:
Will address